### DISCUSSION TITLE
Does CasADI support QCQP problems with qpsol?

### DISCUSSION THREAD

**Discussion by phmandl** [casadi/casadi#3262] [Q&A]
Can I formulate with qpsol a QCQP problem? What happens internally when I provide a nonlinear/quadratic function in g? Optimisation fails at one point for me. I know that ForcesPro supports QCQP. Does CasADI support ForcesPRO? Thanks!

**Comment by jgillis** [comment]
qpsol will linearize g constraints. See https://github.com/casadi/casadi/discussions/3253 
There seems to be an uptick in qcqp request, so maybe I shouls bump priority..
